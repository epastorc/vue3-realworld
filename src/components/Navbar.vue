<template>
  <div class="container-navbar">
    <div>
      <img
        @click="clickAvatar"
        class="avatar"
        src="https://www.blexar.com/avatar.png"
        alt="avatar"
      />
    </div>
    <div>
      <img
        src="https://www.creativefreedom.co.uk/wp-content/uploads/2017/06/Twitter-featured.png"
        alt="twitter logo"
      />
    </div>
    <span></span>
  </div>
</template>

<script lang="ts">
import { defineComponent } from "vue";

    export default defineComponent({
        name:"Navbar",
        props: {
            clickAvatar: Function
        }    
    })
</script>

<style scoped>
body {
  margin: 0px;
}
img {
  height: 40px;
  width: 40px;
  border-radius: 5px;
}
.container-navbar {
  padding: 5px;
  height: 100%;
  display: flex;
  justify-content: space-between;
  align-content: center;
}
.container-navbar > div {
  display: flex;
  align-items: center;
}
</style>