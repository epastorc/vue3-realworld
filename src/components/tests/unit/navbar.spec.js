import { shallowMount } from '@vue/test-utils'
import Navbar from "../../Navbar.vue";

describe('TweetItem.vue', () => {
  it('render component with props', () => {
    const clickAvatar = jest.fn();
      const tweet = {
        accountName: "test",
        avatar:"url",
        content:"content",
        timeAgo:"2ms",
        hashtag:"@test"
      }
    const wrapper = shallowMount(Navbar, {
        props: { clickAvatar: clickAvatar }
      })
    expect(wrapper).toMatchSnapshot();
    wrapper.find(".avatar").trigger("click");
    expect(clickAvatar).toBeCalled();

  })
})
