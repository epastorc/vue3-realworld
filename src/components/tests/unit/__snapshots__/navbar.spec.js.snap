// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TweetItem.vue render component with props 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__emits": Object {
        "0": Object {},
      },
      "__props": Object {
        "0": Array [
          Object {},
          Array [],
        ],
      },
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
      id="app"
    >
      <div
        class="container-navbar"
      >
        <div>
          <img
            alt="avatar"
            class="avatar"
            src="https://www.blexar.com/avatar.png"
          />
        </div>
        <div>
          <img
            alt="twitter logo"
            src="https://www.creativefreedom.co.uk/wp-content/uploads/2017/06/Twitter-featured.png"
          />
        </div>
        <span />
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {},
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__emits": Object {
            "0": null,
          },
          "__props": Object {
            "0": Array [],
          },
          "beforeCreate": [Function],
        },
        Object {
          "__emits": Object {
            "0": null,
          },
          "__props": Object {
            "0": Array [],
          },
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {},
      "reload": [Function],
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.0",
  },
  "__isFunctionalComponent": false,
  "__setProps": [Function],
  "componentVM": Object {
    "__emitted": Object {},
    "clickAvatar": [MockFunction],
  },
  "rootVM": Object {
    "__emitted": Object {},
  },
}
`;
